src/main/java/cloud/fogbow/ras/core/plugins/interoperability/cloudstack/compute/v4_9/CloudStackComputePlugin.java

```diff
@@ -72,33 +85,20 @@ public boolean hasFailed(String cloudState) {
     }
 
     @Override
-    public String requestInstance(ComputeOrder computeOrder, CloudStackUser cloudUser) throws FogbowException {
-        String templateId = computeOrder.getImageId();
-        if (templateId == null || this.zoneId == null || this.defaultNetworkId == null) {
-            LOGGER.error(Messages.Error.UNABLE_TO_COMPLETE_REQUEST);
-            throw new InvalidParameterException();
-        }
+    public String requestInstance(@NotNull ComputeOrder computeOrder, final CloudStackUser cloudUser)
+            throws FogbowException {
 
+        checkCloudUser(cloudUser);
+        String templateId = getTemplateId(computeOrder);
         String userData = this.launchCommandGenerator.createLaunchCommand(computeOrder);
-
-        List<String> networks = new ArrayList<>();
-        networks.add(this.defaultNetworkId);
-        List<String> userDefinedNetworks = computeOrder.getNetworkIds();
-        if (!userDefinedNetworks.isEmpty()) {
-            networks.addAll(userDefinedNetworks);
-        }
-        String networksId = StringUtils.join(networks, ",");
-
+        String networksId = normalizeNetworksID(computeOrder);
+        GetAllServiceOfferingsResponse.ServiceOffering serviceOffering =
+                getServiceOffering(computeOrder, cloudUser);
         int disk = computeOrder.getDisk();
-        GetAllServiceOfferingsResponse.ServiceOffering serviceOffering = getServiceOffering(computeOrder, cloudUser);
         GetAllDiskOfferingsResponse.DiskOffering diskOffering = getDiskOffering(disk, cloudUser);
-        if (serviceOffering == null || diskOffering == null) {
-            throw new NoAvailableResourcesException();
-        }
-
-        String instanceName = computeOrder.getName();
+        String instanceName = normalizeInstanceName(computeOrder.getName());
 
-        DeployVirtualMachineRequest request = new DeployVirtualMachineRequest.Builder()
+        DeployVirtualMachineRequest deployVirtualMachineRequest = new DeployVirtualMachineRequest.Builder()
```

Comment:
Some of the classes used in this plugin are very verbose, in this case, I think the code would be easier to understand if it simplified the name of the variables.
For example _"deployVirtualMachineRequest"_ for _"request"_ only.

Reply:
I will fix this to follow the Fogbow Pattern